consensus/expected.go

```diff
@@ -100,50 +101,154 @@ type Expected struct {
 	genesisCid cid.Cid
 
 	verifier proofs.Verifier
+
+	blockValidator BlockValidator
 }
 
 // Ensure Expected satisfies the Protocol interface at compile time.
 var _ Protocol = (*Expected)(nil)
 
-// NewExpected is the constructor for the Expected consenus.Protocol module.
-func NewExpected(cs *hamt.CborIpldStore, bs blockstore.Blockstore, processor Processor, pt PowerTableView, gCid cid.Cid, verifier proofs.Verifier) Protocol {
-	return &Expected{
-		cstore:       cs,
-		bstore:       bs,
-		processor:    processor,
-		PwrTableView: pt,
-		genesisCid:   gCid,
-		verifier:     verifier,
+type BlockValidator interface {
+	BlockSemanticValidator
+	BlockSyntaxValidator
+	BlockStateValidator
+}
+
+type BlockSyntaxValidator interface {
+	ValidateSyntax(ctx context.Context, ts types.TipSet) error
+}
+
+type BlockSemanticValidator interface {
+	ValidateSemantics(ctx context.Context, cur, parent types.TipSet) error
+}
+
+type BlockStateValidator interface {
+	ValidateState(ctx context.Context, st state.Tree, cur, parent types.TipSet) error
```

Comment:
If this is intended to be something like `RunStateTransition` then I don't think we should try to include it in this validation stuff (yet). In particular it's not clear that this doesn't have side-effects in the current implementation. Limit the scope of change as much as you can.